frontend/src/js/components/settings/PlanSelection.tsx

```diff
@@ -0,0 +1,124 @@
+// Copyright 2021 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+import React, { useEffect } from 'react';
+
+import { Check as CheckIcon, DeveloperBoard as DeveloperBoardIcon } from '@mui/icons-material';
+import { Button } from '@mui/material';
+import { makeStyles } from 'tss-react/mui';
+
+import { AvailablePlans, PLANS, Plan } from '@northern.tech/store/constants';
+import { isEmpty } from '@northern.tech/utils/helpers';
+
+import { clearLocalStorageEntry, isUpgrade } from './Upgrade';
+
+const useStyles = makeStyles()(theme => ({
+  planIcon: { color: theme.palette.primary.main, fontSize: '16px', alignSelf: 'center' },
+  planFeature: { fontSize: '12px' },
+  planButton: { alignSelf: 'center' },
+  planPanel: {
+    padding: '22px 16px',
+    borderRadius: '5px',
+    verticalAlign: 'top',
+    width: '32%',
+    transition: 'all 0.2s ease',
+    cursor: 'pointer',
+    display: 'flex',
+    flexDirection: 'column',
+    justifyContent: 'space-between',
+    marginRight: theme.spacing(2),
+    border: `1px solid ${theme.palette.grey[600]}`,
+    fontSize: '14px',
+    fontWeight: 500,
+    '&.muted, &.muted:hover': {
+      cursor: 'default',
+      boxShadow: 'none'
+    }
+  }
+}));
+const canUpgrade = (plan: AvailablePlans, currentPlan: AvailablePlans) => Object.keys(PLANS).indexOf(plan) > Object.keys(PLANS).indexOf(currentPlan);
+
+const isPlanDisabled = (item: Plan, currentPlan: AvailablePlans, isTrial: boolean, isUpgradePending: Partial<{ plan: AvailablePlans }>) => {
+  return (isUpgradePending.plan && !canUpgrade(item.id, isUpgradePending.plan)) || (!isTrial && !canUpgrade(item.id, currentPlan));
+};
+const isPlanMuted = (item: Plan, currentPlan: AvailablePlans, isTrial: boolean, isUpgradePending: Partial<{ plan: AvailablePlans }>) => {
+  return (
+    isPlanDisabled(item, currentPlan, isTrial, isUpgradePending) &&
+    (item.id !== currentPlan || isTrial) &&
+    !(isUpgradePending.plan && isUpgradePending.plan === item.id)
+  );
+};
```

Comment:
this seems to still have an issue, the following is on a basic plan, with enterprise requested:
<img width="869" alt="Screenshot 2024-12-09 at 11 37 12" src="https://github.com/user-attachments/assets/864e909d-8317-48fb-8389-eb2181c5f0bc">

with the first panel appearing clickable + not muted

Reply:
I'll remove `cursor: pointer` from the `planPanel`. otherwise first panel shouldn't be muted [according to the design](https://www.figma.com/design/0xHrBYtRf2LH9XNjFgjPI3/Billing-contact?node-id=3989-2475&node-type=frame&t=7M5a66Gkgu4pGLQT-0) as it's a current plan